akka-stream/src/main/scala/akka/stream/impl/SplitAfterProcessorImpl.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import scala.util.control.NonFatal
+import akka.actor.Props
+import akka.stream.ActorFlowMaterializerSettings
+import akka.stream.Supervision
+import akka.stream.scaladsl.Source
+
+/**
+ * INTERNAL API
+ */
+private[akka] object SplitAfterProcessorImpl {
+  def props(settings: ActorFlowMaterializerSettings, splitPredicate: Any ⇒ Boolean): Props =
+    Props(new SplitAfterProcessorImpl(settings, splitPredicate))
+
+  private trait SplitDecision
```

Comment:
if we go with two methods `splitWhen` and `splitAfter` I don't think we should also expose `split` with the enums. Otherwise only `split`, but the enums are not so nice (esp for java).